networkx/algorithms/tests/test_distance_measures.py

```diff
@@ -369,22 +390,13 @@ def test_resistance_distance_div0(self):
             self.G[1][2]["weight"] = 0
             nx.resistance_distance(self.G, 1, 3, "weight")
 
-    def test_resistance_distance_not_connected(self):
-        with pytest.raises(nx.NetworkXError):
-            self.G.add_node(5)
-            nx.resistance_distance(self.G, 1, 5)
-
     def test_resistance_distance_same_node(self):
-        with pytest.raises(nx.NetworkXError):
-            nx.resistance_distance(self.G, 1, 1)
+        assert nx.resistance_distance(self.G, 1, 1) == 0
 
-    def test_resistance_distance_nodeA_not_in_graph(self):
-        with pytest.raises(nx.NetworkXError):
-            nx.resistance_distance(self.G, 9, 1)
-
-    def test_resistance_distance_nodeB_not_in_graph(self):
-        with pytest.raises(nx.NetworkXError):
-            nx.resistance_distance(self.G, 1, 9)
+    def test_resistance_distance_all(self):
```

Comment:
Since the semantics of `nodeA` and `nodeB` have been altered, we'll also need additional tests for the new behavior; i.e. a case when `nodeA` is not `None` and another when `nodeB` is no `None`.